src/VisualStudio/Core/Def/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -198,7 +218,8 @@ int IVsSelectionEvents.OnElementValueChanged([ComAliasName("Microsoft.VisualStud
                     ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_Type, out var frameType)) &&
                     (int)frameType == (int)__WindowFrameTypeFlags.WINDOWFRAMETYPE_Document)
                 {
-                    TrackNewActiveWindowFrame(frame);
+                    var runningDocumentTable = ThreadingContext.JoinableTaskFactory.Run(() => GetRunningDocumentTableAsync(ThreadingContext.DisposalToken).AsTask());
```

Comment:
Won't this just block here instead?

Reply:
➡️ In general, no. This method returns immediately in all but the first invocation, and the first invocation _should_ come through the other path. In the event it doesn't it'll wait here for the service to be initialized.

Reply:
I'm not especially worried here because in the off chance there is a problem, it'll be an item on the call stack in RPS results.